virttest/compat.py

```diff
@@ -1,28 +1,66 @@
 import argparse
 
+from avocado.core.settings import settings
 
-def get_opt(opt, name):
-    """
-    Compatibility handler for options in either argparse.Namespace or dict
 
-    :param opt: either an argpase.Namespace instance or a dict
-    :param name: the name of the attribute or key
-    """
-    if isinstance(opt, argparse.Namespace):
-        return getattr(opt, name, None)
-    else:
-        return opt.get(name)
+def is_registering_settings_required():
+    """Checks the characteristics of the Avocado settings API.
 
+    And signals if the explicit registration of options is required, along
+    with other API details that should be followed.
 
-def set_opt(opt, name, value):
-    """
-    Compatibility handler for options in either argparse.Namespace or dict
+    The heuristic used here is to check for methods that are only present
+    in the new API, and should be "safe enough".
 
-    :param opt: either an argpase.Namespace instance or a dict
-    :param name: the name of the attribute or key
-    :param value: the value to be set
+    TODO: remove this once support for Avocado releases before 81.0,
+    including 69.x LTS is dropped.
     """
-    if isinstance(opt, argparse.Namespace):
-        setattr(opt, name, value)
-    else:
+    return all((hasattr(settings, 'add_argparser_to_option'),
+                hasattr(settings, 'register_option'),
+                hasattr(settings, 'as_json')))
+
+
+if is_registering_settings_required():
```

Comment:
Any reason to not move this if inside the methods? I understood the idea here, but sounds a little strange to me, this, since the methods' signatures are the same.

Reply:
I'd rather have it called once, than on every single one of these calls (which are many!).

Reply:
This is a bit of magic indeed but it is mostly due to all the backwards compatibility. Hopefully, it will simplify once the oldest LTS versions are dropped.